### PR TITLE
Implemented a method for transcoding the video through FFMPEG.

### PR DESCRIPTION
**Pull request by ehgoodenough**
## Purpose

To provide utility methods to our developers for transcoding videos.

### CODE REVIEW

**Comment by ehgoodenough** [comment]
selfie-1

**Comment by tmjz402** [comment]
Does it work?

**Comment by ehgoodenough** [comment]
Yes, but will it blend?

File: source/ffmpeg.process.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        {
+            transcoding.audioCodec(audio_codec);
+        }
+        
+        transcoding.on("error", function(error)
+        {
+            reject(error);
+        });
+        
+         * transcoding.on("progress", function(progress)
```
Comment (tmjz402) [inline source/ffmpeg.process.js]: Looks like leftovers from a comment

**Comment by ehgoodenough** [comment]
@tzorca, fixed! I'm so sorry I left those comments in. :<

**Comment by tmjz402** [comment]
It successfully transcoded to ogv and webm, but gave error code 100 when trying to transcode to mp4/libx264.

**Comment by ehgoodenough** [comment]
Hmmmmm! That's really weird. I wonder if it's an issue of codecs, because
each installation of FFMPEG on different platforms use different codecs.

**Comment by ehgoodenough** [comment]
@tzorca: http://www.mediasoftpro.com/error-codes.html

**Comment by ehgoodenough** [comment]
I added some tests as well! :D

**Comment by tmjz402** [comment]
For the test, the relative path is apparently starting from fluent-ffmpeg directory, so it fails to find the asset. Does it work on your machine?

Possibly unhandled Error: ffmpeg exited with code 1: ../assets/kfchvCyHmsc.flv: No such file or directory
    at ChildProcess.<anonymous> (\cumulonimbus\node_modules\fluent-ffmpeg\lib\processor.js:167:22)

**Comment by ehgoodenough** [comment]
I fixed that, but forgot to push it. Try it now! :D

**Comment by tmjz402** [comment]
The tests are working now.

**Comment by ehgoodenough** [comment]
selfie-0

**Pull request merged**